x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -17,58 +16,127 @@ export async function getDocumentSources({
   start,
   end,
   kuery,
+  enableServiceTransactionMetrics,
+  enableContinuousRollups,
 }: {
   apmEventClient: APMEventClient;
   start: number;
   end: number;
   kuery: string;
+  enableServiceTransactionMetrics: boolean;
+  enableContinuousRollups: boolean;
 }) {
-  const sources: Array<ApmDataSource & { hasDocs: boolean }> = flatten(
-    await Promise.all(
-      [
-        ApmDocumentType.ServiceTransactionMetric as const,
-        ApmDocumentType.TransactionMetric as const,
-      ].map(async (documentType) => {
-        const docTypeConfig = getConfigForDocumentType(documentType);
-        const allHasDocs = await Promise.all(
-          docTypeConfig.rollupIntervals.map(async (rollupInterval) => {
-            const response = await apmEventClient.search(
-              'check_document_type_availability',
-              {
-                apm: {
-                  sources: [
-                    {
-                      documentType,
-                      rollupInterval,
-                    },
-                  ],
-                },
-                body: {
-                  track_total_hits: 1,
-                  size: 0,
-                  terminate_after: 1,
-                  query: {
-                    bool: {
-                      filter: [...kqlQuery(kuery), ...rangeQuery(start, end)],
-                    },
-                  },
-                },
-              }
-            );
+  const currentRange = rangeQuery(start, end);
+  const kql = kqlQuery(kuery);
+  const beforeRange = rangeQuery(start - (end - start), end - (end - start));
```

Comment:
I find visualisations helpful but not sure if it just adds noise

```
        before                     current     
|------------------------|------------------------| 
                         |                        |
                      (start)                   (end)
```

Reply:
I'm not sure about the vis here but I did clean up the code a little bit, let me know what you think.